### PR TITLE
Correct NIOHTTPWebClientSocketUpgrader to NIOHTTPWebSocketClientUpgrader

### PR DESCRIPTION
**Pull request by tigerpixel**
Correct 'NIOHTTPWebClientSocketUpgrader' class name to 'NIOHTTPWebSocketClientUpgrader' class name.

### Motivation:

The class name was a typo.
It is a client upgrader that upgrades WebSocket connections, so the updated name suits better.

### Modifications:

Change class name of ‘NIOHTTPWebClientSocketUpgrader’ to ‘NIOHTTPWebSocketClientUpgrader’.
Update the WebSocket client example that uses this class.
Add a typealias to allow the old class name to be used.

### Result:

Improved naming.

### CODE REVIEW

File: Sources/_NIO1APIShims/NIO1APIShims.swift [hunk 1]
```diff
@@ -531,3 +531,6 @@ extension WebSocketFrameDecoder {
         self.init(maxFrameSize: maxFrameSize)
     }
 }
+
+@available(*, deprecated, renamed: "NIOWebSocketClientUpgrader")
+typealias NIOWebClientSocketUpgrader = NIOWebSocketClientUpgrader
```
Comment (Lukasa) [inline Sources/_NIO1APIShims/NIO1APIShims.swift]: This is not a NIO 1 shim: we have to carry this name with us for the rest of the project. It must be imported and available alongside any import of NIOWebSocket. Can you move this into `NIOWebSocketClientUpgrader.swift`?

Comment (tigerpixel) [reply Sources/_NIO1APIShims/NIO1APIShims.swift]: Done.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@weissi is this a known failure mode of the API compatibility tool?

**Comment by weissi** [comment]
@Lukasa hmm, I was pretty sure it handled that one correctly. Maybe we’re using an old version? I think we should repro with 5.2 and if still a problem file a new bug, I filed a few issues already but they’re different

**Comment by Lukasa** [comment]
Yup, I can reproduce this with the Swift 5.2 snapshots, so I think this bug exists upstream. I'll file a bugs.swift.org and see how we go.

**Comment by weissi** [comment]
Oh wow! Thanks for properly checking @Lukasa

**Comment by tigerpixel** [comment]
Nice tooling, done now.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Pull request merged**